java/flight/flight-core/src/main/java/org/apache/arrow/flight/grpc/ServerInterceptorAdapter.java

```diff
@@ -87,10 +87,12 @@ public ServerInterceptorAdapter(List<KeyFactory<?>> factories) {
     // Use LinkedHashMap to preserve insertion order
     final Map<FlightServerMiddleware.Key<?>, FlightServerMiddleware> middlewareMap = new LinkedHashMap<>();
     final MetadataAdapter headerAdapter = new MetadataAdapter(headers);
+    Context currentContext = Context.current();
     for (final KeyFactory<?> factory : factories) {
       final FlightServerMiddleware m;
       try {
         m = factory.factory.onCallStarted(info, headerAdapter);
+        currentContext = m.onAuthenticationSuccess(currentContext);
```

Comment:
Could we find an alternative to changing the context while looping through the factories? Instead of having "onAuthenticationSuccess" as a new function in FlightServerMiddleware, could the logic required here be moved into ServerAuthMiddleware and have this check happen through the normal process of calling "onCallStarted"?

Reply:
The issue here is there's no way to alter the Context from onCallStarted. We can get the context but can't get the ServerInterceptorAdapter to use the context.

Reply:
I created a facade for this and supplied it as an input to onCallStarted.

Reply:
I added another input to onCallStarted() that the implementor can use to write context-specific variable values.